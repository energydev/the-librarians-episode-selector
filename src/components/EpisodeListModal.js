import React from "react";
import Modal from "react-modal";
import { connect } from "react-redux";
import ReactTable from 'react-table';
import { ReactTableDefaults } from 'react-table'
import { guestModalOn, guestModalOff, selectedEpisodeModalOn } from "../actions/modals";
import { setGuest } from "../actions/criteria";

export class EpisodeListModal extends React.Component {

    constructor(props) {
        super(props);

        this.afterOpenModal = this.afterOpenModal.bind(this);

    }

    handleGuestSelection = (e) => {
        console.log("in handleGuestSelection");
        this.props.guestModalOff();
        setTimeout(() => { this.props.selectedEpisodeModalOn(); }, 200);
    };

    onGuestChange = (e) => {
        console.log("in onGuestChange");
        const guest = e.target.value;
        this.props.setGuest(guest);
        setTimeout(() => { console.log(this.props.criteria.guest); }, 200);
    };

    afterOpenModal() {
        console.log("in afterOpenModal");
        this.props.setGuest(this.props.guests[0]);
        setTimeout(() => { console.log(this.props.criteria.guest); }, 200);
    }

    render() {

        return (
            <div className="content-container">
                <Modal
                    isOpen={false}
                    onAfterOpen={this.afterOpenModal}
                    onRequestClose={this.handleGuestSelection}
                    contentLabel="Select a Featured Guest"
                    closeTimeoutMS={200}
                    className="modal"
                >
                    <div className="content-container">
                        <div className="list-header">
                            <div className="show-for-mobile">Episodes</div>
                            <div className="show-for-desktop">Episode</div>
                            <div className="show-for-desktop">Description</div>
                        </div>
                        <div className="list-body">
                            {
                               this.props.episodes.map((episode) => <p key={episode.EpisodeId}>{episode.Description}</p>)
                            }
                        </div>

                    </div>
                    <button className="button" onClick={this.handleGuestSelection}>Okay</button>
                </Modal>
            </div>
        )
    }
};

<option value="Jenkins">Jenkins</option>


const mapStateToProps = (state) => ({
    modals: state.modals,
    criteria: state.criteria
});

const mapDispatchToProps = (dispatch) => ({
    guestModalOff: () => dispatch(guestModalOff()),
    guestModalOn: () => dispatch(guestModalOn()),
    selectedEpisodeModalOn: () => dispatch(selectedEpisodeModalOn()),
    setGuest: (guest) => dispatch(setGuest(guest))
});


export default connect(mapStateToProps, mapDispatchToProps)(EpisodeListModal);