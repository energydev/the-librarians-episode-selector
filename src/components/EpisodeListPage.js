import React from "react";

export const EpisodeListPage = (props) => (
    <div className="episode-list-page">
        <div className="content-container">
            <div className="list-header">
                <div className="show-for-mobile">Episodes</div>
                <div className="show-for-desktop">Episode</div>
                <div className="show-for-desktop">Description</div>
            </div>
            <div className="list-body">
                {
                    props.episodes.map((episode) => {

                        const jsx =
                            (<div className="list-item" key={episode.EpisodeID}>
                                <div className="list-item-main">
                                    <h3 className="list-item__title">{episode.Title}</h3>
                                    <span className="list-item__sub-title">{<p>Season {episode.Season} Episode {episode.EpisodeNoInSeason}</p>}</span>
                                </div>
                                <p className="list-item__data">{episode.Description}</p>
                            </div>
                            );

                        return jsx;
                    })

                }
            </div>
        </div>
    </div>
);

export default EpisodeListPage;