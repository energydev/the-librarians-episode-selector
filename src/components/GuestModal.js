import React from "react";
import Modal from "react-modal";
import { connect } from "react-redux";
import { guestModalOn, guestModalOff, selectedEpisodeModalOn } from "../actions/modals";
import { setGuest } from "../actions/criteria";

export class GuestModal extends React.Component {

    constructor(props) {
        super(props);

        this.afterOpenModal = this.afterOpenModal.bind(this);

    }

    handleGuestSelection = (e) => {
        //console.log("in handleGuestSelection");
        this.props.guestModalOff();
        setTimeout(() => { this.props.selectedEpisodeModalOn(); }, 200);
    };

    onGuestChange = (e) => {
        //console.log("in onGuestChange");
        const guest = e.target.value;
        this.props.setGuest(guest);
        setTimeout(() => { console.log(this.props.criteria.guest); }, 200);
    };

    afterOpenModal() {
        //console.log("in afterOpenModal");
        this.props.setGuest(this.props.guests[0]);
        setTimeout(() => { console.log(this.props.criteria.guest); }, 200);
    }

    render() {
        return (
            <Modal
                isOpen={!!this.props.modals.guest} 
                onAfterOpen={this.afterOpenModal}
                onRequestClose={this.handleGuestSelection}
                contentLabel="Select a Featured Guest"
                closeTimeoutMS={200}
                className="modal"
            >
                <h3 className="modal__title">Guest Selection</h3>
                <div>
                    <select
                        className="input-select"
                        value={this.props.criteria.guest}
                        onChange={this.onGuestChange}
                    >
                        {
                            this.props.guests.map((guest) => (
                                <option key={guest} value={guest}>{guest}</option>
                            ))
                        }
                    </select>
                </div>
                <button className="button" onClick={this.handleGuestSelection}>Okay</button>
            </Modal>
        )
    }
};

<option value="Jenkins">Jenkins</option>


const mapStateToProps = (state) => ({
    modals: state.modals,
    criteria: state.criteria
});

const mapDispatchToProps = (dispatch) => ({
    guestModalOff: () => dispatch(guestModalOff()),
    guestModalOn: () => dispatch(guestModalOn()),
    selectedEpisodeModalOn: () => dispatch(selectedEpisodeModalOn()),
    setGuest: (guest) => dispatch(setGuest(guest))
});


export default connect(mapStateToProps, mapDispatchToProps)(GuestModal);
