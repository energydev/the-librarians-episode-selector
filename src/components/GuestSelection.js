import React from "react";
import { connect } from "react-redux";
import { selectedEpisodeModalOn } from "../actions/modals";
import { setGuest, setSelectBy } from "../actions/criteria";


export class GuestSelection extends React.Component {

    constructor(props) {
        super(props);

        this.props.setGuest(this.props.guests[0]);

    }

    handleGuestSelection = (e) => {
        this.props.setSelectBy("guest");
        setTimeout(() => { this.props.selectedEpisodeModalOn(); }, 200);
    };

    onGuestChange = (e) => {
        const guest = e.target.value;
        this.props.setGuest(guest);
    };

    render() {
        return (
            <div>
                <h3>By Guest</h3>
                <div>
                    <select
                        className="input-select"
                        value={this.props.criteria.guest}
                        onChange={this.onGuestChange}
                    >
                        {
                            this.props.guests.map((guest) => (
                                <option key={guest} value={guest}>{guest}</option>
                            ))
                        }
                    </select>
                </div>
                <button className="button" onClick={this.handleGuestSelection}>Okay</button>
            </div>
        )
    }
};

const mapStateToProps = (state) => ({
    modals: state.modals,
    criteria: state.criteria
});

const mapDispatchToProps = (dispatch) => ({
    selectedEpisodeModalOn: () => dispatch(selectedEpisodeModalOn()),
    setGuest: (guest) => dispatch(setGuest(guest)),
    setSelectBy: (selectBy) => dispatch(setSelectBy(selectBy))
});


export default connect(mapStateToProps, mapDispatchToProps)(GuestSelection);
