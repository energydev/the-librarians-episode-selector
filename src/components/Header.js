import React from "react";
import { Link, withRouter } from "react-router-dom";

export const Header = (props) => (
    <div className="header">
        <div className="content-container">
            <div className="header__content">
            <div>
            <Link className="header__title" to={(location.pathname.includes("the-librarians-episode-selector") ? "/the-librarians-episode-selector/" : "/")}><h1>{props.title}</h1></Link>
            {props.subtitle && <h2 className="header__subtitle">{props.subtitle}</h2>}
            </div>
                <div>
                    <Link className="header__link" to={(location.pathname.includes("the-librarians-episode-selector") ? "/the-librarians-episode-selector/episodes" : "/episodes")}>Episode List</Link>
                </div>
            </div>
        </div>
    </div>
);


Header.defaultProps = {
    title: "Librarians Episode Selector"
}

export default withRouter(Header);
