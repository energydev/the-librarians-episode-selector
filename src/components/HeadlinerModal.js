import React from "react";
import Modal from "react-modal";
import { connect } from "react-redux";
import { headlinerModalOn, headlinerModalOff, selectedEpisodeModalOn } from "../actions/modals";
import { setHeadliner } from "../actions/criteria";


export class HeadlinerModal extends React.Component {

    handleHeadlinerSelection = (e) => {
        console.log("inside handleHeadlinerSelection");
        this.props.headlinerModalOff();
        setTimeout(() => { this.props.selectedEpisodeModalOn(); }, 200);
    };

    onHeadlinerChange = (e) => {
        console.log("inside onHeadlinerChange");
        const headliner = e.target.value;
        this.props.setHeadliner(headliner);
        setTimeout(() => { console.log(this.props.criteria.headliner); }, 200);
    };   

    render() {
        return (
            <Modal
                isOpen={this.props.modals.headliner} //converted to a true boolean
                onRequestClose={this.handleHeadlinerSelection}
                contentLabel="Select a Featured Healiner"
                closeTimeoutMS={200}
                className="modal"
            >
                <h3 className="modal__title">Headliner selection</h3>

                <div>
                <select
                    className="input-select"
                    value={this.props.criteria.headliner}
                    onLoad={this.onHeadlinerChange}
                    onChange={this.onHeadlinerChange}
                >
                    <option value="Flynn">Flynn</option>
                    <option value="Eve">Eve</option>
                    <option value="Cassandra">Cassandra</option>
                    <option value="Ezekiel">Ezekiel</option>
                    <option value="Jake">Jake</option>
                    <option value="Jenkins">Jenkins</option>

                </select>
            </div>
                <button className="button" onClick={this.handleHeadlinerSelection}>Okay</button>
            </Modal>
        )
    }
};


const mapStateToProps = (state) => ({
    modals: state.modals,
    criteria: state.criteria
});

const mapDispatchToProps = (dispatch) => ({
    headlinerModalOff: () => dispatch(headlinerModalOff()),
    headlinerModalOn: () => dispatch(headlinerModalOn()),
    selectedEpisodeModalOn: () => dispatch(selectedEpisodeModalOn()),
    setHeadliner: (headliner) => dispatch(setHeadliner(headliner))
});


export default connect(mapStateToProps, mapDispatchToProps)(HeadlinerModal);
