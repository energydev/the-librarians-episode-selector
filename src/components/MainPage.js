import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import SeasonSelection from "./SeasonSelection";
import SelectedEpisodeModal from "./SelectedEpisodeModal";
import HeadlinerSelection from "./HeadlinerSelection";
import GuestSelection from "./GuestSelection";
import { getGuestCharacters } from "../modules/episodes";
import episodeJson from "../data/LibrariansEpisodes.json";
import Select from 'react-select';

const options = [
    { value: '1', label: 'One' },
    { value: '2', label: 'Two' },
    { value: '3', label: 'Three' },
    { value: '4', label: 'Four' }
];


export class MainPage extends React.Component {

    render() {

        return (
            <div className="main-page">
                <div className="main-page-content-container">
                    <div className="main-page-sub-title">
                        <h2>Randomly Select Episode</h2>
                    </div>
                    <div className="main-page-item">
                        <div className="main-page-section">
                            <SeasonSelection />
                        </div>
                        <div className="main-page-section">
                            <HeadlinerSelection />
                        </div>
                        <div className="main-page-section">
                            <GuestSelection guests={getGuestCharacters(episodeJson.Episodes)} />
                        </div>
                    </div>
                </div>
                <SelectedEpisodeModal />
                />
            </div>
        )
    }
};

const mapStateToProps = (state) => ({
    modals: state.modals
});

export default connect(mapStateToProps)(MainPage);