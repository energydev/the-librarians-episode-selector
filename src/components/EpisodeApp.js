import React from "react";
import Header from "./Header";
import MainPage from "./MainPage";

const subtitle = "Select an Episode to watch";

export const EpisodeApp = (props) => (
    <div className="episode-app">
        <Header subtitle={subtitle} />
        <MainPage />
    </div>
);


export default EpisodeApp;
