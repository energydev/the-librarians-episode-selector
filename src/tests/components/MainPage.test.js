import React from "react";
import { shallow } from "enzyme";
import { MainPage } from "../../components/MainPage";

test("should render MainPage correctly", () => {
    const wrapper = shallow(<MainPage />);
    expect(wrapper).toMatchSnapshot();
});