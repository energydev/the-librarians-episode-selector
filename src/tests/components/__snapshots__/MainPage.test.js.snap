// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainPage correctly 1`] = `
<div
  className="main-page"
>
  <div
    className="main-page-content-container"
  >
    <div
      className="main-page-sub-title"
    >
      <h2>
        Randomly Select Episode
      </h2>
    </div>
    <div
      className="main-page-item"
    >
      <div
        className="main-page-section"
      >
        <Connect(SeasonSelection) />
      </div>
      <div
        className="main-page-section"
      >
        <Connect(HeadlinerSelection) />
      </div>
      <div
        className="main-page-section"
      >
        <Connect(GuestSelection)
          guests={
            Array [
              "Cindy",
              "Dulaque",
              "General Rockwell",
              "Lady of the Lake",
              "Lamia",
              "Mabel",
              "Moriarty",
              "Nicole",
              "Santa Claus",
            ]
          }
        />
      </div>
    </div>
  </div>
  <Connect(SelectedEpisodeModal) />
  /&gt;
</div>
`;
